File: src/main/java/com/example/stopgap/HelidonConfig.java
package com.example.stopgap;

import com.example.stopgap.instanceregistry.Config;

public final class HelidonConfig implements Config {

    private final io.helidon.config.Config source;

    public HelidonConfig(final io.helidon.config.Config source) {
        this.source = source;
    }

    @Override
    public String get(final String key) {
        return source.get(key).asString().orElseThrow();
    }

}

File: src/main/kotlin/com/example/stopgap/HelidonConfig.kt
package com.example.stopgap

import com.example.stopgap.instanceregistry.Config
import io.helidon.config.Config as HeliconC


class HelidonConfig(
    private val source: HeliconC
) : Config {

    override fun get(key: String): String {
        return source.get(key).asString().orElseThrow()
    }

    fun getConfig(key: String): HeliconC {
        return source.get(key)
    }

    companion object {
        fun loadDefault(): HelidonConfig {
            return HelidonConfig(HeliconC.create())
        }
    }
}
